bundles/org.openhab.io.homekit/src/main/java/org/openhab/io/homekit/internal/HomekitTaggedItem.java

```diff
@@ -145,6 +161,78 @@ public HomekitOHItemProxy getProxyItem() {
         return proxyItem;
     }
 
+    /**
+     * Send DecimalType command to a NumberItem (or a Group:Number)
+     * 
+     * @param command
+     * @return true if the item is a NumberItem
+     */
+    public boolean send(DecimalType command) {
+        if (getItem() instanceof GroupItem && getBaseItem() instanceof NumberItem) {
+            ((GroupItem) getItem()).send(command);
+            return true;
+        } else if (getItem() instanceof NumberItem) {
+            ((NumberItem) getItem()).send(command);
+            return true;
+        }
+        return false;
```

Comment:
not sure about return's. 
as far as i saw the return values are not evaluated by the caller. it cannot do much anyway except of print a warning. maybe we could just print warning here. 

if we want caller to take care, then we should consider throwing an exception. 
not sure whether we have a openHAB guidance on usage of return values vs exceptions

same applies to other send methods.

Reply:
I see where you're coming from. My original intention was to make it so the caller could avoid the type check itself, by doing something like

1)
```
if (send(percentType)) {
   return;
} else if (send(decimalType)) {
  return;
} else {
  logger.warn(...)
}
```

(which I now see could even simplify to, even if a bit opaque):

2)
```
if (!send(percentType) && !send(decimalType)) {
  logger.warn(...);
}
```

But then I ended up not doing it because each of those cases require constructing a new object for the command based on the type, so it would be an unnecessary allocation if it's not that type. I'm open to going any direction here
 * doing 1) above
 * doing 2) above
 * returning void (assume the caller already did the type check)
 * raising an exception in the false case (assume the caller already did the type check)

Reply:
im just afraid that at some point the developer implementing the caller will forget to check the return value and nothing will happen in case of wrong type. 

i just checked how such cases are handled in the OH core and GenericItem for instance just print a warning
https://github.com/openhab/openhab-core/blob/6758aacba59478938281249a6ca05265717de772/bundles/org.openhab.core/src/main/java/org/openhab/core/items/GenericItem.java#L443

i would suggest we do the same, just add warning logging before "return false" in the send methods. if the caller done correctly and checks the type before calling "send" we would never see this warning.